### PR TITLE
[GSoC] preferences refactors + move "Open changelog" to the "About" screen

### PR DESCRIPTION
**Pull request by BrayanDSO**
## Purpose / Description
cleanup

## Approach
On the commits

### AnkiDroid directory

https://user-images.githubusercontent.com/69634269/184417791-506bee3b-b810-477e-b76c-6ae90e65fcfd.mp4

### Language

https://user-images.githubusercontent.com/69634269/184417654-4b0e19ef-764f-4707-bed2-7874ff68a256.mp4

### Changelog

https://user-images.githubusercontent.com/69634269/184417628-7476704e-f325-4fe6-a054-783a8d91b6d5.mp4

## Checklist
_Please, go through these checks before submitting the PR._

- [X] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [X] You have a descriptive commit message with a short title (first line, max 50 chars).
- [X] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [X] You have commented your code, particularly in hard-to-understand areas
- [X] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

**Comment by BrayanDSO** [comment]
The changes are diverse, but the PR size itself is relatively small, so I've thought that doing them all in one PR would be best (and avoid more self-conflicts to myself). If there is a problem about this approach, I can split the PR.

**Review by lukstbit — APPROVED** [review]
Ok! Wandering if this url shouldn't be extracted to a constant:

https://github.com/ankidroid/Anki-Android/blob/02097062f288640d68335c3d14e9eab7079d13ce/AnkiDroid/src/main/java/com/ichi2/anki/preferences/AdvancedSettingsFragment.kt#L69

**Comment by BrayanDSO** [comment]
> Wandering if this url shouldn't be extracted to a constant 

As it is only used in one place, I think it's fine either way

File: AnkiDroid/src/main/java/com/ichi2/anki/Preferences.kt [hunk 1]
```diff
@@ -123,6 +124,21 @@ class Preferences : AnkiActivity() {
     // Class methods
     // ----------------------------------------------------------------------------
 
+    /**
+     * Enables and sets the visibility of the "Developer options" header on [HeaderFragment]
+     */
+    fun enableDevOptions(isEnabled: Boolean) {
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/Preferences.kt:130]: rename: `setDevOptionsEnabled`.

You don't want passing 'false' to 'enable' to cause it to be disabled.

**Review by david-allison — CHANGES_REQUESTED** [review]
Reiterating the main question:

Can you ensure you test this heavily with a corrupt collection, or when opening the collection throws an exception. A user needs to be able to access the screen even if the opening fails

File: AnkiDroid/src/main/java/com/ichi2/anki/preferences/AdvancedSettingsFragment.kt [hunk 2]
```diff
@@ -46,6 +46,7 @@ class AdvancedSettingsFragment : SettingsFragment() {
                 val newPath = newValue as String?
                 try {
                     CollectionHelper.initializeAnkiDroidDirectory(newPath)
```
[the remark this answers is not in the document]
Comment (BrayanDSO) [reply AnkiDroid/src/main/java/com/ichi2/anki/preferences/AdvancedSettingsFragment.kt:48]: As I've discussed [here](https://github.com/ankidroid/Anki-Android/pull/11873#discussion_r925027037), I can't reproduce the error handling path that leads to opening directly the advanced screen.

But anyway, I don't see how this is related to the problem. The only change this does is that previously, to change the directory you had to `Tap AnkiDroid directory > Change the directory > Tap ok > Back press OR tap the appBar menu button`. The commit only removes the last step, i.e. now is `Tap AnkiDroid directory > Change the directory > Tap ok`

**Review by david-allison — APPROVED** [review]

**Comment by lukstbit** [comment]
@BrayanDSO Merging the other preferences related PR introduced a conflict. Could you fix it so we can merge this PR as well?

**Comment by BrayanDSO** [comment]
Oh, these merge conflicts... 🤔

image

**Review by lukstbit — APPROVED** [review]

**Pull request merged**